common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -809,6 +809,10 @@ public static enum ConfVars {
     HIVE_IN_TEST_REPL("hive.in.repl.test", false, "internal usage only, true in replication test mode", true),
     HIVE_IN_TEST_IDE("hive.in.ide.test", false, "internal usage only, true if test running in ide",
         true),
+    HIVE_IN_TEST_PLANMAPPER_STRICT_VALIDATION("hive.in.test.planmapper.strict.validation", false,
+        "internal use only, whether to raise an error when unexpected links are found. We ignore equivalence mapping "
+        + "violation because it introduces only minor problems. But we want to strictly check it in qtest so that we "
+        + "can prevent further degradations"),
```

Comment:
I am unsure about this newly introduced property. It may prevent future "degradation" but it puts an extra burden on contributors and reviewers.

As a reviewer, when I see people setting on/off properties in tests I always ask why? And if the answer is "because there is a problem" then I will ask for more information. The contributor will try to understand what's going on and eventually it will find this ticket and argue that this feature is broken so its best to disable the property for their use-case (as out of scope). The reviewer may agree or push back and say that this is blocking the PR from merging.

An exception is by definition serious, a warning on the other hand is not so much and I feel that for this use-case it should have been a warning in the first-place.

Reply:
Thanks. Let's say our qtest covers 99.9% of use cases of Hive(I trust the coverage 😃 ), which means 0.1% are uncovered. I assume we agreed that it is too costly to achieve 100% promptly.
I personally thought it'd be more reasonable to use CI if we wanted to keep the stability high enough, e.g. 90%. I'm a bit pessimistic here and I'm going to remove this property.

Reply:
Done by https://github.com/apache/hive/pull/5452/commits/4f60470184e766798f6bf32319aca90c97446312